src/smartHandler.ts

```diff
@@ -195,10 +202,23 @@ export class SMARTHandler implements Authorization {
 
         if (fhirUserObject) {
             const { hostname, resourceType, id } = fhirUserObject;
+
             if (isFhirUserAdmin(fhirUserObject, this.adminAccessTypes, fhirServiceBaseUrl)) {
                 // if an admin do not add limiting search filters
                 return [];
             }
+
+            references.add(`${hostname}/${resourceType}/${id}`);
+            if (hostname === fhirServiceBaseUrl) {
+                references.add(`${resourceType}/${id}`);
+            }
+            if (request.resourceType && request.resourceType === resourceType) {
```

Comment:
Can you remove the duplicated lines 226-228?

Reply:
did you mean make the following repeated code as separate function?
```
            references.add(`${hostname}/${resourceType}/${id}`);
            if (hostname === fhirServiceBaseUrl) {
                references.add(`${resourceType}/${id}`);
            }
            if (request.resourceType && request.resourceType === resourceType) {
                ids.add(id);
            }
```

Reply:
No. It looks to me like lines 226-228 are redundant. They are already covered by lines 212-216.

Reply:
there are three condition check here: `fhirUserObject`, `patientOrgs`, and `patientLaunchContext` which adds references and ids. These are redundant similar codes but the check condition is independent of each other. Please let me know if I am missing something here.

Reply:
My apologies. The diff is sometimes hard to read. I thought these 2 blocks of the same code were in the same scope. I see now that they aren't. Disregard my comment.